### PR TITLE
 msg list: Download file on clicking file url.

### PR DESCRIPTION
**Pull request by jainkuniya**

### CODE REVIEW

**Comment by gnprice** [comment]
Thanks @jainkuniya ! Quick comments (copying what I told you today in PMs and audio):
* This needs more explanation. What was the old behavior; what have you learned about it; what changes in the behavior with your code changes; and why are those changes the right ones?
* I think #M3487 is related. Discussing that relationship would be especially helpful. (Perhaps it's even exactly the issue this fixes?)
* Also please fix the build :slightly_smiling_face:

**Comment by borisyankov** [comment]
This is implementing #3303

File: src/utils/url.js [hunk 1]
```diff
@@ -172,3 +172,9 @@ export const autocompleteUrl = (value: string = '', protocol: string, append: st
     : '';
 
 export const isValidUrl = (url: string): boolean => urlRegex({ exact: true }).test(url);
+
+export const isUrlFilePath = (url: string) =>
```
Comment (borisyankov) [inline src/utils/url.js]: This is certainly not robust enough.
For example, `mysite.com/welcome.php` does match it but should not be downloaded.

Usually, the distinction between a downloadable file and viewable file is made based on the mime type, and often the server can 'enforce' (or maybe more correctly 'suggest') the downloadability of a file.

**Comment by borisyankov** [comment]
It is not clear to me if the behavior, even implemented perfectly, is desirable. This does not match the behavior of serval chat apps I did test, including Slack

**Comment by borisyankov** [comment]
I consider this a better (though different) solution to the same problem #3495

File: src/utils/url.js (hunk 1 above)
Comment (jainkuniya) [reply src/utils/url.js]: Yup! I agree it's not robust.

But to get ` mime type` (from header) we need to do a fetch (send a request), and let's say if file is 1mb, then first 1mb (and time) is gone for checking ` mime type` and if it is downloadable file then download it (again 1mb + time).

**Comment by gnprice** [comment]
Cool, thanks @borisyankov for the review and the debugging on #3303 , and thanks @jainkuniya for the updates.

I agree with Boris that this isn't the right direction for a fix; we'll continue on the issue #3303 and on #3495 .

**Pull request closed without merging**